dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -128,6 +128,12 @@ def year
       application_year
     end
 
+    # Since teacher applications may be created on save before they are submitted, we cannot rely
+    # on the created_at field. When applications are submitted, they have status 'unreviewed'
+    def date_applied
+      status_log.find {|status_entry| status_entry["status"] == "unreviewed"}&.[]("at")&.to_date&.iso8601
```

Comment:
I hacked this together ... suggestions for making this easier to read and cleaner are welcome!

Reply:
looks good! the unit test makes it clear what this is doing 👍